### ISSUE TITLE
[Feature] Google Pay Integration

### ISSUE THREAD

**Issue by dominikponniah** [admin-ch/CovidCertificate-App-Android#175]
Google Pay provides the possibility to use the COVID-Certificate inside the App. Is there any plan fo this integration (maybe automatically)? 

See: https://support.google.com/pay/answer/10890261

Maybe this would also be an approach to do so on Apple Pay / Wallet-App? By using integrated Shortcuts like Buttons or Direct-Payment-Features the Code could be opened easier and faster.

**Comment by Hen-Rex** [comment]
The feature has been, as of yet, released in the US only, as stated in your link.

**Comment by dominikponniah** [comment]
@Hen-Rex i know; but maybe a Mark in the Backlog would be Great so that the feature can be used as soon as released by Google. Alternatively a simple .pkpass-File should also Work for this Usecase :-)

**Comment by 10Meisterbaelle** [comment]
There is an API for COVID Cards: https://developers.googleblog.com/2021/06/google-updates-passes-api-to-save-covid-testing-and-vaccination-information-on-Android-devices.html?m=1

**Comment by 10Meisterbaelle** [comment]
> Alternatively a simple .pkpass-File should also Work for this Usecase :-)
No, because Google Pay doesn't support .pkpass

**Comment by dominikponniah** [comment]
You're right, but there is the Passes API: https://developers.google.com/pay/passes/guides/pass-verticals/use-cases?vertical=covid-cards#pass-vertical and of course the simplest way to provide the .pkpass-File - this would then work on similar Apps like WalletPasses - or maybe Apps like Samsung-Pay?

I experienced for example at Blue Cinemas (Swisscom) and also at Europa Park (MackInternational GmbH und Co. KG) that these provide simple pkpass-Files for Android-Users too.

**Comment by dmengelt** [comment]
any update on this?

**Comment by DRSchlaubi** [comment]
> The feature has been, as of yet, released in the US only, as stated in your link.

It seems like it has reached europe now, as spain has implemented it

**Comment by DRSchlaubi** [comment]
> I experienced for example at Blue Cinemas (Swisscom) and also at Europa Park (MackInternational GmbH und Co. KG) that these provide simple pkpass-Files for Android-Users too.

There are 3rd party apps, that can act as a pkpass wallet on android and there also is pass2pay so you can add pkpass files to gpay, but none of them work 100% like the passes api itself

**Comment by dominikponniah** [comment]
@DRSchlaubi thats true, but Google Released a new API at I/O this month. Using a Validated Organization-Account (the Bund should have enough Money to get a Developers License for 25$...) you can simply create a WHO-Vaccine-Pass.

There are also simple Webapps which are able to create Passes: https://coronapass.fabianpimminger.com/

As Google wants to transform its Pay-App back to the Wallet, there may be also minor support for pkpass-Files or a abstracted standard.

_TbH the Pass-App by CH-Gov is the worst App i've ever seen. _

**Comment by DRSchlaubi** [comment]
> thats true, but Google Released a new API at I/O this month. Using a Validated Organization-Account (the Bund should have enough Money to get a Developers License for 25$...) 

It's true that they can afford that, but they don't want to support Google Pay

> There are also simple Webapps which are able to create Passes: https://coronapass.fabianpimminger.com/

All of these websites don't support GPay except for an Austrialian one iirc, which doesn't support EU certs

> As Google wants to transform its Pay-App back to the Wallet, there may be also minor support for pkpass-Files or a abstracted standard.

There is a new API which doesn't require the certs being stored on Google Servers, but you still need to be a verified Pay business to use it and only officials can verify as a business with access to the covid certificates
